src/openzeppelin/tests/test_accesscontrol.cairo

```diff
@@ -2,124 +2,308 @@ use openzeppelin::access::accesscontrol::AccessControl;
 use openzeppelin::utils::constants::{DEFAULT_ADMIN_ROLE, IACCESSCONTROL_ID};
 use starknet::{contract_address_const, ContractAddress, testing::set_caller_address};
 
-const SOME_OTHER_ROLE: felt252 = 42;
+const ROLE: felt252 = 41;
+const OTHER_ROLE: felt252 = 42;
 
-fn ACCOUNT1() -> ContractAddress {
+fn ADMIN() -> ContractAddress {
     contract_address_const::<1>()
 }
 
-fn ACCOUNT2() -> ContractAddress {
+fn AUTHORIZED() -> ContractAddress {
     contract_address_const::<2>()
 }
 
+fn OTHER() -> ContractAddress {
+    contract_address_const::<3>()
+}
+
+fn OTHER_ADMIN() -> ContractAddress {
+    contract_address_const::<4>()
+}
+
 fn setup() {
-    set_caller_address(ACCOUNT1());
-    AccessControl::constructor(ACCOUNT1());
+    AccessControl::_grant_role(DEFAULT_ADMIN_ROLE, ADMIN());
+    set_caller_address(ADMIN());
 }
 
+//
+// initializer
+//
+
 #[test]
 #[available_gas(2000000)]
 fn test_initializer() {
     AccessControl::initializer();
     assert(AccessControl::supports_interface(IACCESSCONTROL_ID), 'Should support own interface');
 }
 
+//
+// has_role
+//
+
 #[test]
 #[available_gas(2000000)]
-fn test_constructor() {
-    AccessControl::constructor(ACCOUNT1());
-    assert(AccessControl::has_role(DEFAULT_ADMIN_ROLE, ACCOUNT1()), 'Admin role should be set');
-    assert(AccessControl::supports_interface(IACCESSCONTROL_ID), 'Should support own interface');
+fn test_has_role() {
+    setup();
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'should not have role');
+    AccessControl::_grant_role(ROLE, AUTHORIZED());
+    assert(AccessControl::has_role(ROLE, AUTHORIZED()), 'should have role');
 }
 
+
+//
+// assert_only_role
+//
+
 #[test]
 #[available_gas(2000000)]
 fn test_assert_only_role() {
     setup();
-    AccessControl::assert_only_role(DEFAULT_ADMIN_ROLE);
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    set_caller_address(AUTHORIZED());
+    AccessControl::assert_only_role(ROLE);
 }
 
 #[test]
 #[available_gas(2000000)]
 #[should_panic(expected: ('Caller is missing role', ))]
-fn test_assert_only_role_unauthorized() {
+fn test_assert_only_role_unauthorized_1() {
     setup();
-    set_caller_address(ACCOUNT2());
-    AccessControl::assert_only_role(DEFAULT_ADMIN_ROLE);
+    set_caller_address(OTHER());
+    AccessControl::assert_only_role(ROLE);
 }
 
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Caller is missing role', ))]
+fn test_assert_only_role_unauthorized_2() {
+    setup();
+    set_caller_address(AUTHORIZED());
+    AccessControl::assert_only_role(OTHER_ROLE);
+}
```

Comment:
what was the rationale to add this extra test?

Reply:
This was to test that granting ROLE to AUTHORIZED won't allow AUTHORIZED to pass as other roles. It was missing the grant `ROLE` to `AUTHORIZED`

Reply:
let's make this clear in the test name then